node/core/pvf/src/prepare/worker.rs

```diff
@@ -314,59 +307,61 @@ async fn recv_request(stream: &mut UnixStream) -> io::Result<(Vec<u8>, PathBuf,
 /// The entrypoint that the spawned prepare worker should start with. The `socket_path` specifies
 /// the path to the socket used to communicate with the host.
 pub fn worker_entrypoint(socket_path: &str) {
-	worker_event_loop("prepare", socket_path, |mut stream| async move {
+	worker_event_loop("prepare", socket_path, |stream| async move {
+		let mutex = Arc::new(Mutex::new((stream, false)));
 		loop {
-			let (code, dest, preparation_timeout) = recv_request(&mut stream).await?;
-
+			let (code, dest, preparation_timeout) = {
+				let mut lock = mutex.lock().await;
+				// Unset the lock flag. We set it when either thread finishes.
+				lock.1 = false;
```

Comment:
This creates a race condition. 

If you finish the job(timeout hasn't fired yet), restart the loop, timeout fires and takes down worker.

Reply:
The polling thread doesn't terminate the worker, it sends the timeout error to the pool, which in turn terminates the process. Another worker is spawned for the next task.

Should be fine?

Reply:
Do we always start a new worker? Don't we reuse workers?

Reply:
We do reuse them when they conclude with `rip: false`.
A timeout means that `rip: true` and it should be restarted.

But after re-reading your comment I see where the race is, agreed.
I believe the worker should simply `join` the timeout thread after successfully completing the task, that should do.

Reply:
Sorry for the race, I need to be more careful. 😬